src/main/java/io/phdata/streamliner/schemadefiner/ConfigBuilder/ScriptCommand.java

```diff
@@ -0,0 +1,193 @@
+                            map.put("configuration", configuration);
+                            map.put("table", hadoopTable);
+                            map.put("typeMapping", typeMapping);
+                            rendered = JavaHelper.getLayout(templateFile.getPath(), map);
+                          }
+
+                          String replaced = rendered.replace("    ", "\t");
+                          log.debug(replaced);
+                          String fileName =
+                              tableDir + "/" + templateFile.getName().replace(".ssp", "");
+                          StreamlinerUtil.writeFile(replaced, fileName);
+                          StreamlinerUtil.isExecutable(fileName);
+                        });
+
+                nonTemplateFiles.stream()
+                    .forEach(
+                        nonTemplateFile -> {
+                          String content = StreamlinerUtil.readFile(nonTemplateFile.getPath());
+                          String fileName = tableDir + "/" + nonTemplateFile.getName();
+                          StreamlinerUtil.writeFile(content, fileName);
+                          StreamlinerUtil.isExecutable(fileName);
+                        });
+              });
+      writeSchemaMakeFile(configuration, typeMapping, templateDirectory, outputDirectory);
+    } else if (configDiff != null) {
+      log.info(
+          "Configuration Difference (example: streamliner-configuration-diff.yml) file found.");
+      if (configDiff.getTableDiffs() == null) {
+        throw new RuntimeException(
+            "TableDiff section is not found. Check the configuration difference (example: streamliner-configuration-diff.yml) file");
+      }
+      TemplateContext templateContext = new TemplateContext();
+      configDiff.getTableDiffs().stream()
+          .forEach(
+              tableDiff -> {
+                String tableDiffDir = outputDirectory + "/" + tableDiff.getDestinationName();
+                StreamlinerUtil.createDir(tableDiffDir);
+                String template = templateDirectory + "/" + pipeline + "/" + "alter-table.sql.ssp";
+                File templateFile = new File(template);
+
+                log.info("Rendering file: {} ", templateFile);
+                Map<String, Object> map = new LinkedHashMap<>();
+                map.put("configurationDiff", configDiff);
+                map.put("tableDiff", tableDiff);
+                map.put("typeMapping", typeMapping);
+                map.put("templateContext", templateContext);
+                String rendered = JavaHelper.getLayout(templateFile.getPath(), map);
+                String replaced = rendered.replace("    ", "\t");
+                log.debug(replaced);
+                String str = replaced.replaceAll("\n", "").replaceAll("\t", "");
+                String fileName = tableDiffDir + "/" + templateFile.getName().replace(".ssp", "");
+                if (str.length() != 0) {
```

Comment:
When does this happen? Maybe a good place for a log message?

Reply:
Initially I was handling Drop Table and create a new table as well. In case there is change in column like drop column or modify column but no table is created or dropped then it was creating an empty file. To restrict that I added this check. But now since we are only handling ALTER TABLE ADD COLUMN this should not happen. I will remove this.

Reply:
removed